core/include/cubos/core/net/utils.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#ifdef _WIN32
+#include <winsock2.h>
+#else
+#include <sys/socket.h>
+#endif
+
+#include <string>
+
+// Socket type. Differs from Windows (SOCKET) and POSIX (int).
+using InnerSocket = decltype(socket(0, 0, 0));
+
+// Represents an invalid socket.
+constexpr InnerSocket InnerInvalidSocket = static_cast<InnerSocket>(-1);
```

Comment:
Should both be documented, and marked as being for internal use. They should also be within the namespace.